specifyweb/frontend/js_src/lib/components/LocalityUpdate/types.ts

```diff
@@ -0,0 +1,101 @@
+import type { State } from 'typesafe-reducer';
+
+import type { IR, RA } from '../../utils/types';
+import type { Tables } from '../DataModel/types';
+
+export type LocalityUpdateHeader = Exclude<
+  Lowercase<
+    | keyof Tables['GeoCoordDetail']['fields']
+    | keyof Tables['Locality']['fields']
+  >,
+  'locality'
+>;
+
+export type LocalityUpdateParseError = {
+  readonly message: string;
+  readonly field: string;
+  readonly payload: IR<unknown>;
+  readonly rowNumber: number;
+};
+
+export type LocalityUpdateTaskStatus =
+  | 'ABORTED'
+  | 'FAILED'
+  | 'PARSE_FAILED'
+  | 'PARSED'
+  | 'PARSING'
+  | 'PENDING'
+  | 'PROGRESS'
+  | 'SUCCEEDED';
+
+export type LocalityUpdateState =
+  | State<
+      'ABORTED',
+      { readonly taskstatus: 'ABORTED'; readonly taskinfo: string }
```

Comment:
hmm, this duplicates the fields
the `State<>` helper adds a `{state:'ABORTED'}`, and then you redundantly add `taskstatus`
Ideally you would pick one or another
If you want to keep the taskstatus name, then get rid of `State<>`